File: ChatGPT/Direct_Attempt/python/output/output_15a.py
# ChatGPT-generated solution will go here.
def main():
    with open("input.txt") as f:
        lines = [line.rstrip('\n') for line in f]

    grid = []
    moves = ""
    reading_grid = True

    for line in lines:
        if reading_grid:
            if line.strip() == "":
                reading_grid = False
            else:
                grid.append(list(line))
        else:
            moves += line.strip()

    rows = len(grid)
    cols = len(grid[0])
    directions = {'^': (-1, 0), 'v': (1, 0), '<': (0, -1), '>': (0, 1)}

    # Locate robot and boxes
    for r in range(rows):
        for c in range(cols):
            if grid[r][c] == '@':
                robot_pos = (r, c)
                grid[r][c] = '.'

    for move in moves:
        dr, dc = directions[move]
        nr, nc = robot_pos[0] + dr, robot_pos[1] + dc

        if not (0 <= nr < rows and 0 <= nc < cols):
            continue

        target = grid[nr][nc]
        if target == '.':
            robot_pos = (nr, nc)
        elif target == 'O':
            nnr, nnc = nr + dr, nc + dc
            if 0 <= nnr < rows and 0 <= nnc < cols and grid[nnr][nnc] == '.':
                grid[nnr][nnc] = 'O'
                grid[nr][nc] = '.'
                robot_pos = (nr, nc)

    total = 0
    for r in range(rows):
        for c in range(cols):
            if grid[r][c] == 'O':
                total += r * 100 + c

    print(total)

if __name__ == "__main__":
    main()

File: ChatGPT/Direct_Attempt/clojure/output/output_15a.clj
;; ChatGPT-generated solution will go here.
(ns output_15a
  (:require [clojure.string :as str]
            [clojure.set :as set]))

(defn parse-input [lines]
  (let [[grid-lines move-lines] (split-with #(not (str/starts-with? % "<")) lines)
        grid (vec (map vec grid-lines))
        moves (->> move-lines
                   (apply str)
                   (remove #{\newline})
                   (map {\^ [-1 0] \v [1 0] \< [0 -1] \> [0 1]}))]
    {:grid grid :moves moves}))

(defn find-positions [grid ch]
  (for [r (range (count grid))
        c (range (count (first grid)))
        :when (= (get-in grid [r c]) ch)]
    [r c]))

(defn in-bounds? [grid [r c]]
  (and (<= 0 r) (< r (count grid))
       (<= 0 c) (< c (count (first grid)))))

(defn walk [state]
  (let [{:keys [grid moves]} state
        robot (first (find-positions grid \@))
        boxes (set (find-positions grid \O))]
    (loop [pos robot
           boxes boxes
           moves moves]
      (if (empty? moves)
        {:final-boxes boxes}
        (let [[dr dc] (first moves)
              next-pos [(+ dr (first pos)) (+ dc (second pos))]]
          (cond
            (not (in-bounds? grid next-pos)) (recur pos boxes (rest moves))
            (= (get-in grid next-pos) \#) (recur pos boxes (rest moves))
            (boxes next-pos)
            (let [beyond [(+ dr (first next-pos)) (+ dc (second next-pos))]]
              (if (and (in-bounds? grid beyond)
                       (not= (get-in grid beyond) \#)
                       (not (boxes beyond)))
                (recur next-pos
                       (-> boxes (disj next-pos) (conj beyond))
                       (rest moves))
                (recur pos boxes (rest moves))))
            :else (recur next-pos boxes (rest moves))))))))

(defn gps [[r c]] (+ (* 100 r) c))

(defn part [lines]
  (let [{:keys [grid moves]} (parse-input lines)
        {:keys [final-boxes]} (walk {:grid grid :moves moves})]
    (reduce + (map gps final-boxes))))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
